src/context/WalletProvider/MetaMask/components/Connect.tsx

```diff
@@ -90,6 +92,13 @@ export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
             return dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
           const isSnapInstalled = await checkIsSnapInstalled()
 
+          const snapVersion = await getSnapVersion()
+
+          const isCorrectVersion = snapVersion === getConfig().REACT_APP_SNAP_VERSION
```

Comment:
Why don't we use the hook and do `const { isSnapInstalled, isCorrectVersion } = useIsSnapInstalled()` on init of `MetaMaskConnect` instead here?

Reply:
It *should* be doable though the reason why we may not have done it previously is probably too high of reactivity - will give it a try!

Reply:
FYI @0xApotheosis checked this one out and we can't do this because of provider shenanigans it looks like